src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -17,7 +16,7 @@ namespace Microsoft.FeatureManagement
     /// <summary>
     /// Used to evaluate whether a feature is enabled or disabled.
     /// </summary>
-    class FeatureManager : IFeatureManager, IFeatureVariantManager
+    class FeatureManager : IFeatureManager, IDynamicFeatureManager
```

Comment:
I'm thinking whether we should split the implementation of two interfaces into separate classes. They are pretty much independent. This could be confusing for someone new to look at this code in the future. Thoughts?

Reply:
I don't see the value immediately. What is the confusion that you had in mind?

Reply:
I guess the question can be asked in the other way - why do we implement both interfaces in one class? Would we do the same if we started from scratch? The functionalities of the two interfaces are parallel. It's not like one is a complement to another. In current naming, feels like the FeatureManager is really for IFeatureManager, and the IDynamicFeatureManager is a supplement (like IDisposable).

Reply:
> Would we do the same if we started from scratch?

I think that's a great way to look at it. I'd say we'd have them separated. Let me separate them.

Reply:
> In current naming, feels like the FeatureManager is really for IFeatureManager, and the IDynamicFeatureManager

For this point, it would be different if we renamed `IFeatureManager` to `IFeatureFlagManager`. But that was a breaking change that we didn't consider.

Reply:
> For this point, it would be different if we renamed IFeatureManager to IFeatureFlagManager. But that was a breaking change that we didn't consider.

Totally, that would make more sense. Another option is to add what's in IDynamicFeatureManager to IFeatureManager directly. I think that's what we would do if we started from scratch, but we worried it will break all existing libraries built on top of the current feature management library.

Reply:
Yes, to be honest I'm not completely sold on either side. If you think it'll be more clear I'm happy to separate them. Otherwise, can keep in in `FeatureManager`

Reply:
I think we should separate the implementation if we decide to separate the interface.

Reply:
In that case I'll go ahead and separate ConfigurationFeatureDefinitionProvider and FeatureManagerSnapshot as well.

Reply:
Updated.